### PR TITLE
Implanting revheads and gang bosses actually uses up the implant.

### PR DESCRIPTION
**Pull request by harmonyn**
Fixes https://github.com/tgstation/tgstation/issues/28629.

### CODE REVIEW

**Comment by QualityVan** [comment]
Should probably change the failure message to indicate the implant is used up.

**Review by Cyberboss — CHANGES_REQUESTED** [review]
https://github.com/tgstation/tgstation/blob/master/code/game/objects/items/weapons/implants/implant.dm#L44

This will destroy the implant in EVERY failure case. e.g. Target already has an implant, is not suitable for implants, etc.

**Comment by bgobandit** [comment]
oops

**Comment by bgobandit** [comment]
@Cyberboss as far as I can tell this needs to be refactored to be fixed, my options are afaict

1) Move the checks for can_be_implanted and already has implant out of implant() and into implanter's attack, so they just return before the whole nonsense starts
2) Add special code in the loyalty implant proc to handle this
3) Add some kind of var to be returned as well to govern whether the implant is trashed
4) Call this intended behavior -- you're sticking it in, why wouldn't the implant be trashed

**Comment by Cyberboss** [comment]
Have the implant delete itself if the conditions are correct, then check QDELETED(imp) in implanter/attack

**Comment by bgobandit** [comment]
>stale

**Review by Cyberboss — CHANGES_REQUESTED** [review]
You still need to make the implant delete itself in implant/implant()

**Comment by bgobandit** [comment]
the qdel happens in loyalty implant code, it was already there

File: code/game/objects/items/weapons/implants/implanter.dm [hunk 1]
```diff
@@ -37,10 +37,11 @@
 						to_chat(user, "<span class='notice'>You implant yourself.</span>")
 					else
 						M.visible_message("[user] has implanted [M].", "<span class='notice'>[user] implants you.</span>")
-					imp = null
```
Comment (Cyberboss) [inline code/game/objects/items/weapons/implants/implanter.dm]: Restore this line

File: code/game/objects/items/weapons/implants/implanter.dm [hunk 2]
```diff
@@ -37,10 +37,11 @@
 						to_chat(user, "<span class='notice'>You implant yourself.</span>")
 					else
 						M.visible_message("[user] has implanted [M].", "<span class='notice'>[user] implants you.</span>")
-					imp = null
-					update_icon()
 				else
 					to_chat(user, "<span class='warning'>[src] fails to implant [M].</span>")
+				if(QDELETED(imp))
+					imp = null
+					update_icon()
```
Comment (Cyberboss) [inline code/game/objects/items/weapons/implants/implanter.dm]: untab this line

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by KorPhaeron** [comment]
Stale

**Pull request closed without merging**